From af42a10576a0af50423c7f55d5b5cf3e58bf334f Mon Sep 17 00:00:00 2001
From: Andrew Savchenko <bircoph@gentoo.org>
Date: Sun, 12 Dec 2021 01:06:18 +0300
Subject: [PATCH 14/14] kernel-cc-ld

Pass custom $CC and $LD to kernel during both configure check and build.
Otherwise hosts without `gcc` and `ld` wrappers will fail configure.

See Gentoo bug 828951.
---
 src/cf/linux-test1.m4             | 4 ++--
 src/libafs/MakefileProto.LINUX.in | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff -u -r -N a/src/cf/linux-test1.m4 b/src/cf/linux-test1.m4
--- a/src/cf/linux-test1.m4	2023-07-06 10:43:20.000000000 -0400
+++ b/src/cf/linux-test1.m4	2024-03-12 07:35:03.724533585 -0400
@@ -24,8 +24,8 @@
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
 _ACEOF
-    echo make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD &&
-    make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD 2>conftest.err &&
+    echo make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 CC="${CC}" LD="${LD}" >&AS_MESSAGE_LOG_FD &&
+    make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 CC="${CC}" LD="${LD}" >&AS_MESSAGE_LOG_FD 2>conftest.err &&
     ! grep -i "WARNING: .* undefined!$" conftest.err >/dev/null 2>&1
     then [$3]
     else
diff -u -r -N a/src/libafs/MakefileProto.LINUX.in b/src/libafs/MakefileProto.LINUX.in
--- a/src/libafs/MakefileProto.LINUX.in	2023-07-06 10:43:20.000000000 -0400
+++ b/src/libafs/MakefileProto.LINUX.in	2024-03-12 07:35:31.864221423 -0400
@@ -322,7 +322,7 @@
 ${LINUX_LIBAFS_NAME}.ko afspag.ko: .FORCE
 	env EXTRA_CFLAGS="${EXTRA_CFLAGS}" @TOP_SRCDIR@/libafs/make_kbuild_makefile.pl ${KDIR} $@ @TOP_OBJDIR@/src/config/Makefile.config Makefile.afs Makefile.common
 	( env EXTRA_CFLAGS="${EXTRA_CFLAGS}" \
-		$(MAKE) -C ${LINUX_KERNEL_BUILD} M=@TOP_OBJDIR@/src/libafs/${KDIR} $(SPARSE_MAKEFLAGS) modules 2>&1 \
+		$(MAKE) -C ${LINUX_KERNEL_BUILD} M=@TOP_OBJDIR@/src/libafs/${KDIR} $(SPARSE_MAKEFLAGS) CC="${CC}" LD="${LD}" modules 2>&1 \
 		|| echo "FAILURE: make exit code $$?" ) | tee .makelog
 	@if grep -E -q '^(FAILURE|WARNING)' .makelog; then \
 		grep -q '^WARNING' .makelog && echo "Error: Undefined symbols in modules" ; \
